pkg/cstorpool/v1alpha2/cstorpool.go

```diff
@@ -69,30 +88,80 @@ func (l *cspList) FilterUIDs(p ...predicate) []string {
 // listBuilder enables building a
 // list of csp instances
 type listBuilder struct {
-	list *cspList
+	list *CSPList
 }
 
 // ListBuilder returns a new instance of
 // listBuilder object
 func ListBuilder() *listBuilder {
-	return &listBuilder{list: &cspList{}}
+	return &listBuilder{list: &CSPList{}}
 }
 
 // WithUIDs builds a list of cstor pools
 // based on the provided pool UIDs
 func (b *listBuilder) WithUIDs(poolUIDs ...string) *listBuilder {
 	for _, uid := range poolUIDs {
-		obj := &apis.CStorPool{}
-		obj.SetUID(types.UID(uid))
-		item := &csp{object: obj}
-		b.list.items = append(b.list.items, item)
+		obj := &CSP{}
+		obj.Object.SetUID(types.UID(uid))
+		b.list.Items = append(b.list.Items, obj)
+	}
+	return b
+}
+
+// WithUIDNodeMap builds a cspList based on the provided
+// map of uid and nodename
+func (b *listBuilder) WithUIDNodeMaps(UIDNodeMaps ...map[string]interface{}) *listBuilder {
+	for _, UIDNodeMap := range UIDNodeMaps {
+		for k, v := range UIDNodeMap {
+			obj := &CSP{}
+			obj.Object.SetUID(types.UID(k))
+			obj.Object.SetAnnotations(map[string]string{string(scheduleOnHostAnnotation): v.(string)})
+			b.list.Items = append(b.list.Items, obj)
+		}
 	}
 	return b
 }
 
+// WithCStorPool builds the list based on the provided
+// *apis.CStorPool instances
+func (b *listBuilder) WithCStorPoolList(pools *CSPList) *listBuilder {
+	b.list = pools
+	return b
+}
+
 // List returns the list of csp
 // instances that were built by
 // this builder
-func (b *listBuilder) List() *cspList {
+func (b *listBuilder) List() *CSPList {
 	return b.list
 }
+
+// GetPoolUIDs retuns the UIDs of the pools
+// available in the list
+func (l *CSPList) GetPoolUIDs() []string {
+	uids := []string{}
+	for _, pool := range l.Items {
+		uids = append(uids, string(pool.Object.GetUID()))
+	}
+	return uids
+}
+
+// buildCSPListFromUIDNodeMaps exposes WithUIDNodeMap
+// to CAS Templates
+func buildCSPListFromUIDNodeMaps(UIDNodeMap map[string]interface{}) *CSPList {
+	return ListBuilder().WithUIDNodeMaps(UIDNodeMap).List()
+}
+
+// buildCSPListFromUIDS exposes WithUIDs to CASTemplates
+func buildCSPListFromUIDs(uids []string) *CSPList {
+	return ListBuilder().WithUIDs(uids...).List()
+}
+
+// TemplateFunctions exposes a few functions as go
+// template functions
+func TemplateFunctions() template.FuncMap {
+	return template.FuncMap{
+		"createCSPListFromUIDs":       buildCSPListFromUIDs,
+		"createCSPListFromUIDNodeMap": buildCSPListFromUIDNodeMaps,
```

Comment:
Rename the template functions accordingly

Reply:
done